src/hooks/useActiveWeb3React.ts

```diff
@@ -17,12 +19,48 @@ const isActive = (web3Data: Web3Data): web3Data is ActiveWeb3Data =>
   !!web3Data.library &&
   !!web3Data.connector;
 
+export const useEagerConnect = () => {
+  const { activate, active } = useWeb3React(); // specifically using useWeb3React because of what this hook does
+  const [tried, setTried] = useState(false);
+
+  useEffect(() => {
+    injectedConnector.isAuthorized().then((isAuthorized) => {
```

Comment:
Maybe let's change it to `async/await` and `try+catch` to make it look more consistent to other hooks in our app?